File: CovarianzaYcontraVarianza_C/Bicicleta.cc
#include"Bicicleta.h"
#include<iostream>
#include<string>
using namespace std;

Bicicleta::Bicicleta(string p,int r):Movilidad(p){
  tamRuedas = r;
}

void Bicicleta::avanzar(){
  cout<<"La bicicleta de "<<propietario<<" con ruedas de tamanio "<<tamRuedas<<"cm. Avanza!"<<endl;
}

File: Covarianza y contravarianza java/src/to/Bicicleta.java
package to;

public class Bicicleta extends Movilidad implements Rueda{
	protected int numRuedas;
	
	public Bicicleta(){}
	
	public Bicicleta(String propietario, int numRuedas) {
		super(propietario);
		this.numRuedas = numRuedas;
	}

	@Override
	public void avanzar() {
		System.out.println("La Bicicleta avanza!");
	}

	@Override
	public void detener() {
		System.out.println("La Bicicleta se detuvo!");	
	}
	
	

}
